OpenLaTeXStudio/Editor/src/main/java/latexstudio/editor/EditorTopComponent.java

```diff
@@ -276,6 +287,51 @@ public void commentOutText() {
         }
 
     }
+    
+    public void spellCheckAllText() throws BadLocationException {     
+        Document doc = rSyntaxTextArea.getDocument();   
+        if (doc != null) {                                                
+            String editorText = rSyntaxTextArea.getText(0, doc.getLength());            
+            if (editorText != null) {  
+                Highlighter highlighter = rSyntaxTextArea.getHighlighter();
+                Highlighter.HighlightPainter painter = new DefaultHighlighter.DefaultHighlightPainter(Color.PINK);  //Default color is: PINK
+
+                JLanguageTool langTool = new JLanguageTool(new AmericanEnglish());  //Default language is: American English
+                for (Rule rule : langTool.getAllActiveRules()) {
+                    if (rule instanceof SpellingCheckRule) {
+                        ((SpellingCheckRule)rule).acceptPhrases(Arrays.asList("tex", "Tex", "TEX", "documentclass", "maketitle"));  //Accept some TEX terms
+                    }
+                }
+                
+                boolean highlighted = getEditorState().isHighlighted();
+                if(!highlighted) {  //If highlight isn't enabled before clicking
+                        List<RuleMatch> matches = null;  
+
+                        try {
+                            matches = langTool.check(editorText);
+                        } catch (IOException ex) {
+                            Exceptions.printStackTrace(ex);
+                        }
+
+                        //Highlight the spelling check results
+                        for (RuleMatch match : matches) {
+                            highlighter.addHighlight(match.getFromPos(), match.getToPos(), painter);   
+                        }                        
+                } else {  //If highlight is already enabled before clicking
+                    Highlighter.Highlight[] highlighters = highlighter.getHighlights();                
+                    for (int i = 0; i < highlighters.length; i++) {

```

Comment:
There's a method called "removeAllHighlights". Perhaps it's enough to call it, instead of manually iterating through all highlights?